Source/KerbalConstructionTime/KCTGameStates.cs

```diff
@@ -80,6 +81,23 @@ public static void InitAndClearTechList()
                 TechList = new KCTObservableList<TechItem>();
             TechList.Updated += KerbalConstructionTime.Instance.UpdateTechlistIconColor;
         }
+
+        public static void CreateNewPad(string padName, int padLevel)
+        {
+            KCT_LaunchPad lp = ActiveKSC.ActiveLPInstance;
+            List<UpgradeableFacility> list = lp.GetUpgradeableFacilityReferences();
+            var upgdFacility = list[0];
+            UpgradeableObject.UpgradeLevel[] padUpgdLvls = upgdFacility.UpgradeLevels;
+
+            padLevel = UnityEngine.Mathf.Clamp(padLevel, 1, padUpgdLvls.Length);
+
+            ActiveKSC.LaunchPads.Add(new KCT_LaunchPad(padName, padLevel));
```

Comment:
Probably should do some safety checks here, rather than risking NREs?

```
if (lp.GetUpgradeableFacilityReferences()?.FirstOrDefault()?.UpgradeLevels is UpgradeableObject.UpgradeLevel[] padUpgdLvls)
{
    padLevel = UnityEngine.Matf.Clamp(padLevel, 1, padUpgdLvls.Length)
    ActiveKSC.LaunchPads.Add(new KCT_LaunchPad(padName, padLevel)
}
```

Reply:
resolved by 548608c